src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -84,9 +85,29 @@ internal sealed class ImmutabilityContext {
 				return true;
 			}
 
-			if( TryGetImmutableTypeInfo( type, out ImmutableTypeInfo info ) ) {
-				if( info.Kind.HasFlag( kind ) ) {
-					return true;
+			if( type is INamedTypeSymbol namedType ) {
+				if( TryGetImmutableTypeInfo( namedType, out ImmutableTypeInfo info ) ) {
+					if( info.Kind.HasFlag( kind ) ) {
+						if( namedType.TypeArguments.Length != info.ImmutableTypeParameters.Length ) {
+							// could this happen?

```

Comment:
🤷

Reply:
I don't think so, `throw new Exception()` is fine.